### PR TITLE
Deflake AlluxioMasterProcessTest

### PR DESCRIPTION
**Pull request by jenoudet**
After looking online, it appears that using PowerMockito with parameterized tests can be unstable (e.g. [link](https://stackoverflow.com/questions/2454473/junit-parameterized-tests-together-with-powermock-how)). Since this test does not require PowerMockito, I used Mockito instead.
I also refactored the `startStopTest` so that I would only wait on the services that it expects to be started. This speeds up the test execution.

### CODE REVIEW

**Comment by tcrain** [comment]
Can you try with this as false please https://github.com/Alluxio/alluxio/blob/master/core/common/src/test/java/alluxio/master/PortRegistry.java#L77

File: core/server/common/src/main/java/alluxio/master/MasterProcess.java [hunk 1]
```diff
@@ -187,9 +188,32 @@ void registerServices(GrpcServerBuilder serverBuilder,
    * @return whether the grpc server became ready before the specified timeout
    */
   public boolean waitForGrpcServerReady(int timeoutMs) {
+    return pollFor(this + " to start", this::isGrpcServing, timeoutMs);
```
Comment (tcrain) [inline core/server/common/src/main/java/alluxio/master/MasterProcess.java:191]: the isGrpcServing function has data races on mGrpcServer and mStarted in GrpcServer.java. These need to be volatile or atomic boolean/pointer

Comment (jenoudet) [reply core/server/common/src/main/java/alluxio/master/MasterProcess.java:191]: made appropriate variables volatile.

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java [hunk 2]
```diff
@@ -152,12 +150,11 @@ public boolean isInSafeMode() {
   }
 
   @Override
-  @Nullable
   public InetSocketAddress getWebAddress() {
-    if (mWebServer != null) {
+    if (isWebServing()) {
```
Comment (tcrain) [inline core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:154]: There could also be a worry for a null pointer exception on mWebServer on line 155, depending on where it is called. Also according to the interface this is overriding it should return null when not serving.

Comment (jenoudet) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: I changed the interface: the callers never check for null.

**Review by tcrain — APPROVED** [review]

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**